### DISCUSSION TITLE
Unexpected errors and warnings when running under firejail

### DISCUSSION THREAD

**Discussion by fjnewt** [netblue30/firejail#5632] [Q&A] [answered]
**Using firejail & firetools 0.9.72 on PCLOS:** 
When run 'firejail opera' or any app from command line the output shows what appears to be errors and though Opera browser opens, it does not open directly to private window as previous FJ versions would have done. A few lines in example of output. (there are lots more)

When open any app from firetools, they do open but is everything fine - I cannot say. Except of course Opera which I expect to open with a private window but does not.

```
Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Warning: not remounting /var/run/user/500/gvfs
Child process initialized in 129.98 ms
[13:13:0129/185504.647212:ERROR:content_main_runner_impl.cc(473)] Unable to load CDM /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so (error: /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so: failed to map segment from shared object)
[1:1:0129/185504.647238:ERROR:content_main_runner_impl.cc(473)] Unable to load CDM /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so (error: /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so: failed to map segment from shared object)
[6:29:0129/185504.732408:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/firejail/mnt/dbus/system: Permission denied
[6:29:0129/185504.732480:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/firejail/mnt/dbus/system: Permission denied
[6:36:0129/185504.742362:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/firejail/mnt/dbus/system: Permission denied
[6:6:0129/185504.874346:ERROR:object_proxy.cc(623)] Failed to call method: org.kde.KWallet.isEnabled: object_path= /modules/kwalletd5: org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
[6:6:0129/185504.874417:ERROR:kwallet_dbus.cc(100)] Error contacting kwalletd5 (isEnabled)
[6:6:0129/185504.875221:ERROR:object_proxy.cc(623)] Failed to call method: org.kde.KLauncher.start_service_by_desktop_name: object_path= /KLauncher: org.freedesktop.DBus.Error.ServiceUnknown: The name org.kde.klauncher was not provided by any .service files
[6:6:0129/185504.875234:ERROR:kwallet_dbus.cc(72)] Error contacting klauncher to start kwalletd5
[6:6:0129/185505.003237:ERROR:object_proxy.cc(623)] Failed to call method: org.kde.KWallet.close: object_path= /modules/kwalletd5: org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
[6:6:0129/185505.003258:ERROR:kwallet_dbus.cc(418)] Error contacting kwalletd5 (close)
WARNING: radv is not a conformant Vulkan implementation, testing use only.
MESA-INTEL: warning: Haswell Vulkan support is incomplete
libGL error: MESA-LOADER: failed to open crocus: /usr/lib64/dri/crocus_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib64/dri, suffix _dri)
libGL error: failed to load driver: crocus
libGL error: MESA-LOADER: failed to open crocus: /usr/lib64/dri/crocus_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib64/dri, suffix _dri)
libGL error: failed to load driver: crocus

```

The same type of output with Firefox -it does not open to a private window as I do not have it set that way.

firejail thunderbird output, a few lines of what the output looks like these (they repeat a lot):

```
Warning: not remounting /var/run/user/500/gvfs
Seccomp list in: !chroot, check list: @default-keep, prelist: unknown,
Child process initialized in 133.73 ms
(thunderbird:9): dconf-CRITICAL **: 19:00:08.367: unable to create file '/var/run/user/500/dconf/user': Read-only file system.  dconf will not work properly.
(thunderbird:9): dconf-CRITICAL **: 19:00:08.367: unable to create file '/var/run/user/500/dconf/user': Read-only file system.  dconf will not work properly.

```

What do these output mean?

**Comment by kmk3** [comment]
(Offtopic)

Please see the following links for how to format code blocks in markdown:

* <https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks>
* <https://github.github.com/gfm/#fenced-code-blocks>

**Reply by fjnewt** [reply]
It is likely I posted this issue in the wrong spot or asked the wrong question or even shouldn't have posted at all. Yet, I won't know for sure unless someone from FJ who knows/cares reply. I believe I followed @kmk3's suggestion re formatting code blocks. 

So does my issue with FJ deserves a response or not?   Thank you.

**Reply by rusty-snake** [reply, marked as answer]
Note that edits do not create notifications.

I for myself ignore the most issues/question that miss basic information. Just wasted to much time on some users.

To explain the errors:

> Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.

Comes from firejail.

If you

- do enable dbus restriction
- do not restrict abstract Unix sockets (protocol or net)
- and have abstract dbus socket in your system,

you will see this. 99% of the firejail users get this. 

It is a warning of a sandbox hole but nothing that indicates breakage.

> Warning: not remounting /var/run/user/500/gvfs

Comes from firejail.

Quite uncommon but as long as you do not need to access gvfs it shouldn't be a problem.

> ```
> [13:13:0129/185504.647212:ERROR:content_main_runner_impl.cc(473)] Unable to load CDM /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so (error: /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so: failed to map segment from shared object)
> [1:1:0129/185504.647238:ERROR:content_main_runner_impl.cc(473)] Unable to load CDM /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so (error: /home/xxx/.config/opera/WidevineCdm/4.10.2557.0/_platform_specific/linux_x64/libwidevinecdm.so: failed to map segment from shared object)
> [6:29:0129/185504.732408:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/firejail/mnt/dbus/system: Permission denied
> [6:29:0129/185504.732480:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/firejail/mnt/dbus/system: Permission denied
> [6:36:0129/185504.742362:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/firejail/mnt/dbus/system: Permission denied
> [6:6:0129/185504.874346:ERROR:object_proxy.cc(623)] Failed to call method: org.kde.KWallet.isEnabled: object_path= /modules/kwalletd5: org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
> [6:6:0129/185504.874417:ERROR:kwallet_dbus.cc(100)] Error contacting kwalletd5 (isEnabled)
> [6:6:0129/185504.875221:ERROR:object_proxy.cc(623)] Failed to call method: org.kde.KLauncher.start_service_by_desktop_name: object_path= /KLauncher: org.freedesktop.DBus.Error.ServiceUnknown: The name org.kde.klauncher was not provided by any .service files
> [6:6:0129/185504.875234:ERROR:kwallet_dbus.cc(72)] Error contacting klauncher to start kwalletd5
> [6:6:0129/185505.003237:ERROR:object_proxy.cc(623)] Failed to call method: org.kde.KWallet.close: object_path= /modules/kwalletd5: org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
> [6:6:0129/185505.003258:ERROR:kwallet_dbus.cc(418)] Error contacting kwalletd5 (close)
> ```

Comes from Opera/Chromium.

All are known/expected with the preconfigured profile.

> ```
> WARNING: radv is not a conformant Vulkan implementation, testing use only.
> MESA-INTEL: warning: Haswell Vulkan support is incomplete
> libGL error: MESA-LOADER: failed to open crocus: /usr/lib64/dri/crocus_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib64/dri, suffix _dri)
> libGL error: failed to load driver: crocus
> libGL error: MESA-LOADER: failed to open crocus: /usr/lib64/dri/crocus_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib64/dri, suffix _dri)
> libGL error: failed to load driver: crocus
> ```

Comes from Opera/Chromium or one of its 3p libraries.

Either you changed the profile, or you have a strange GPU setup or experimental opera flags in use.

PCLOS uses SysVinit AFAIK. Firejail, as the most other software too, sees a lot less testing outside of the homogeneous systemd set (Debian, Ubuntu, Mint, Fedora, Arch, OpenSUSE, Manjaro, Rocky, CentOS, ...) and sometimes has trouble if things work differently.

It looks like `$XDR_RUNTIME_DIR` is under `/var` on you system. If that is the case, try to add:

```
writable-var
```

**Reply by fjnewt** [reply]
Thank you for replying. I do appreciate it.
